backend/command/application/product/product_service.go

```diff
@@ -0,0 +1,78 @@
+package product
+
+import (
+	"errors"
+
+	entities "github.com/Lazaro-Barros/buteco/command/domain/entities/product"
+)
+
+// ProductService lida com os casos de uso para produtos.
+type ProductService struct {
+	repo entities.ProductRepository
+}
+
+// NewProductService cria um novo ProductService.
+func NewProductService(repo entities.ProductRepository) *ProductService {
+	return &ProductService{repo: repo}
+}
+
+// CreateProduct é o caso de uso para criar um novo produto.
+func (s *ProductService) CreateProduct(
+	name string,
+	description string,
+	value int,
+	visible bool,
+) (uuid string, err error) {
+	product, err := entities.NewProduct(name, description, value, visible)
+	if err != nil {
+		return "", err
+	}
+
+	err = s.repo.Save(product)
+	if err != nil {
+		return "", err
+	}
+
+	return product.Uuid(), nil
+}
+
+// UpdateProduct é o caso de uso para atualizar um produto existente.
+func (s *ProductService) UpdateProduct(
+	uuid string,
+	name string,
+	description string,
+	value int,
+	visible bool,
+) error {
+	_, err := s.repo.GetByUuid(uuid)
+	if err != nil {
+		return errors.New("product not found")
+	}
```

Comment:
Essa condição não faz sentido estar dentro do GetByUuid? para não precisar repetir isso toda vez que for procurar por um produto.

Reply:
pode sim, mas ai a gente estaria eliminando uma funcionalidade do getByID, se em alguma situação, a gente quiser obter o uuid, e se o produto nao existir e a gente nao quiser que ocorra erro? tipo, pode ter algum cenario, que o produto existir não seja um erro em si, so uma possibilidade.